### PR TITLE
FIX: Memory leak + infinity future

### LINKED ISSUE #14: Infinity future on close when WebSocket connection will throw
If something will go wrong with establishing connection in DeepgramLiveTranscriber error will be throw leaving us with memory leak as `_outputTranscriptStream`. Calling close will never reach _outputTranscriptStream.close() since future will stay infinitely on `_wsChannel.sink.close(status.normalClosure)` because there is no available sink.

Same goes with infinitely future for close, if we will try to await for it it will never end since e.g.  if `_outputTranscriptStream` has no listeners, closing will never return;

### PR DESCRIPTION
**Pull request by PcolBP**
Described issue in #14 .

Added getter for isClosed to check whenever connection was closed.

Closes #14

### CODE REVIEW

**Comment by PcolBP** [comment]
I've noticed also that you upgrade minimum version of web_socket_channel, but not every package has upgraded it as well, wouldn't it better to support various versions? 👀  I'll provide additional commit for that, let me know if thats ok.

**Comment by PcolBP** [comment]
Previously it was >2.4.0 so to maintain previous version, I've provided '>=2.4.0 <4.0.0' up to version 4 since it may have some braking changes in future.

**Comment by tempo-riz** [comment]
hey @PcolBP thanks for the PR,

isClosed should start at true, that's correct, adding a try/catch around isReady is also a good idea !

The getter is handy aswell

Tell me if i'm wrong but I think there is no need for _hasInitializationException since if start() throws, isClosed is never set to false so calling close would do nothing : 
```
Future<void> close() async {
    if (_isClosed) return;
...
```

**Comment by PcolBP** [comment]
We have to close as well controller so since it will leave all the time if not closed.

I'll change isClosed on true, you're. 

If we would like to remove _hasInitializationException we should make make controller initialisation later or make it nullable from the start 🤔

**Comment by tempo-riz** [comment]
ah yes you are right, maybe we could make _outputTranscriptStream a late field, and instantiate it after a succesful _wsChannel.ready ?

**Comment by PcolBP** [comment]
It depends on the users use case but obviously it would be the breaking change since behaviour would change

**Comment by tempo-riz** [comment]
yeah I would like to keep the API is it is, all things considered your changes look good, but I'm not sure that close() would work as expected

**Comment by tempo-riz** [comment]
Never mind LGTM. only thing is _isClosed should be instantiated to true, since by default the stream is open

File: lib/src/deepgram.dart [hunk 1]
```diff
@@ -14,7 +14,13 @@ class DeepgramLiveTranscriber {
   DeepgramLiveTranscriber(this.apiKey,
       {required this.inputAudioStream, this.queryParams});
 
-  /// your Deepgram API key
+  /// Flag which determines if transcriber was closed
+  bool _isClosed = true;
```
Comment (tempo-riz) [inline lib/src/deepgram.dart:18]: this should stay false

Comment (PcolBP) [reply lib/src/deepgram.dart]: Changed on false 🫡

**Pull request merged**